File: example-servers/python/flask/src/services/basic.py
from flask import Response
import time
import json

class Basic:
  @staticmethod
  def create_chat_body(body, stream=False):
    chat_body = {
      'messages': [
        {
          'role': 'assistant' if message['role'] == 'ai' else message['role'],
          'content': message['text']
        } for message in body['messages']
      ],
      'model': body['model']
    }
    if stream:
      chat_body['stream'] = True
    return chat_body


  def chat(self, body):
    print(body)
    # sends response back to Deep Chat using the Result format:
    # https://deepchat.dev/docs/connect/#Result
    return {'result': {'text': 'This is a respone from a Flask server. Thankyou for your message!'}}


  def chat_stream(self, body):
    print(body)
    response_chunks = 'This is a response from a Flask server. Thank you for your message!'.split(' ')
    response = Response(self.send_stream(response_chunks), mimetype='text/event-stream')
    response.headers['Content-Type'] = 'text/event-stream'
    response.headers['Cache-Control'] = 'no-cache'
    response.headers['Connection'] = 'keep-alive'
    response.headers['Access-Control-Allow-Origin'] = '*'
    return response

  def send_stream(self, response_chunks, chunk_index=0):
    if chunk_index < len(response_chunks):
        chunk = response_chunks[chunk_index]
        yield f"data: {json.dumps({'result': {'text': f'{chunk} '}})}\n\n"
        time.sleep(0.07)
        yield from self.send_stream(response_chunks, chunk_index + 1)
    else:
        yield ''


  def files(self, request):
    # files are stored inside a files object
    files = request.files.getlist('files')
    if files:
      print('Files:')
      for file in files:
        print(file.filename)

      # when sending text messages along with files - they are stored inside the data form
      textMessages = list(request.form.items())
      if len(textMessages) > 0:
        print('Text messages:')
        for key, value in textMessages:
          print(key, value)
    else:
      # when sending text messages without any files - they are stored inside a json
      print("Text messages:")
      print(request.json)

    # sends response back to Deep Chat using the Result format:
    # https://deepchat.dev/docs/connect/#Result
    return {'result': {'text': 'This is a respone from a Flask server. Thankyou for your message!'}}

File: example-servers/node/express/src/services/basic.ts
import {Request, Response} from 'express';

export class Basic {
  public static async chat(body: Request['body'], res: Response) {
    // Text messages are stored inside request body using the Deep Chat JSON format:
    // https://deepchat.dev/docs/connect
    console.log(body);
    // Sends response back to Deep Chat using the Result format:
    // https://deepchat.dev/docs/connect/#Result
    res.json({result: {text: 'This is a respone from ExpressJs server. Thankyou for your message!'}});
  }

  public static async chatStream(body: Request['body'], res: Response) {
    // Text messages are stored inside request body using the Deep Chat JSON format:
    // https://deepchat.dev/docs/connect
    console.log(body);
    const responseChunks = 'This is a respone from ExpressJs server. Thankyou for your message!'.split(' ');
    res.setHeader('Content-Type', 'text/event-stream');
    res.setHeader('Cache-Control', 'no-cache');
    res.setHeader('Connection', 'keep-alive');
    res.setHeader('Access-Control-Allow-Origin', '*');
    Basic.sendStream(res, responseChunks);
  }

  private static sendStream(res: Response, responseChunks: string[], chunkIndex = 0) {
    setTimeout(() => {
      const chunk = responseChunks[chunkIndex];
      if (chunk) {
        // Sends response back to Deep Chat using the Result format:
        // https://deepchat.dev/docs/connect/#Result
        res.write(`data: ${JSON.stringify({result: {text: `${chunk} `}})}\n\n`);
        Basic.sendStream(res, responseChunks, chunkIndex + 1);
      } else {
        res.end();
      }
    }, 70);
  }

  public static async files(req: Request, res: Response) {
    // Files are stored inside a form using Deep Chat request FormData format:
    // https://deepchat.dev/docs/connect
    if (req.files as Express.Multer.File[]) {
      console.log('Files:');
      console.log(req.files);
    }
    // When sending text along with files, they are stored inside the request body using the Deep Chat JSON format:
    // https://deepchat.dev/docs/connect
    if (Object.keys(req.body).length > 0) {
      console.log('Text messages:');
      console.log(req.body);
    }
    // Sends response back to Deep Chat using the Result format:
    // https://deepchat.dev/docs/connect/#Result
    res.json({result: {text: 'This is a respone from ExpressJs server. Thankyou for your message!'}});
  }
}
